src/nodes/branch.rs

```diff
@@ -63,7 +63,10 @@ impl Decodable for BranchNode {
             // Decode without advancing
             let Header { payload_length, .. } = Header::decode(&mut &bytes[..])?;
             let len = payload_length + length_of_length(payload_length);
-            stack.push(RlpNode::from_raw_rlp(&bytes[..len])?);
+            stack.push(RlpNode::from_raw_rlp(&bytes[..len]).map_err(|e| match e {
+                TrieError::RlpNodeTooLarge { .. } => alloy_rlp::Error::Custom("RLP node too large"),
+                _ => alloy_rlp::Error::Custom("unexpected error decoding RLP node"),
+            })?);
```

Comment:
seems like this change can be reverted since the conversion `From<TrieError> for alloy_rlp::Error` was implemented